api/server/sdk/cluster_domains.go

```diff
@@ -0,0 +1,131 @@
+	req *api.SdkClusterDomainsEnumerateRequest,
+) (*api.SdkClusterDomainsEnumerateResponse, error) {
+	if m.cluster() == nil {
+		return nil, status.Error(codes.Unavailable, "Resource has not been initialized")
+	}
+
+	resp, err := m.cluster().Enumerate()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Cannot enumerate cluster cluster domains: %v", err)
+	}
+	clusterResp := &api.SdkClusterDomainsEnumerateResponse{}
+
+	for clusterDomainName, _ := range resp.ClusterDomainsActiveMap {
+		clusterResp.ClusterDomainNames = append(clusterResp.ClusterDomainNames, clusterDomainName)
+	}
+	return clusterResp, nil
+}
+
+func (m *ClusterDomainsServer) Inspect(
+	ctx context.Context,
+	req *api.SdkClusterDomainInspectRequest,
+) (*api.SdkClusterDomainInspectResponse, error) {
+	if m.cluster() == nil {
+		return nil, status.Error(codes.Unavailable, "Resource has not been initialized")
+	}
+
+	if len(req.GetClusterDomainName()) == 0 {
+		return nil, status.Errorf(codes.InvalidArgument, "Must provide a valid cluster domain name")
+	}
+
+	resp, err := m.cluster().Enumerate()
+	if err != nil {
+		return nil, status.Errorf(codes.Internal, "Cannot enumerate cluster cluster domains: %v", err)
+	}
+
+	for clusterDomainName, isActive := range resp.ClusterDomainsActiveMap {
+		if clusterDomainName == req.GetClusterDomainName() {
+			return &api.SdkClusterDomainInspectResponse{
+				ClusterDomainName: clusterDomainName,
+				IsActive:          isActive,
+			}, nil
+		}
+	}
+	return nil, status.Errorf(codes.InvalidArgument, "Cannot find a cluster domain with name: %v", req.GetClusterDomainName())
+}
+
+func (m *ClusterDomainsServer) Activate(
+	ctx context.Context,
+	req *api.SdkClusterDomainActivateRequest,
+) (*api.SdkClusterDomainActivateResponse, error) {
+	if m.cluster() == nil {
+		return nil, status.Error(codes.Unavailable, "Resource has not been initialized")
+	}
+
+	if !api.IsAdminByContext(ctx) {
+		return nil, status.Errorf(codes.PermissionDenied, "Activating cluster domain %v not permitted", req.GetClusterDomainName())
+	}
+
+	if len(req.GetClusterDomainName()) == 0 {
+		return nil, status.Errorf(codes.InvalidArgument, "Must provide a valid cluster domain name")
+	}
+
+	if err := m.cluster().ActivateClusterDomain(&api.ActivateClusterDomainRequest{
+		ClusterDomain: req.GetClusterDomainName(),
+	}); err != nil {
+		return nil, status.Errorf(codes.Internal, "Failed to activate cluster domain %v: %v", req.GetClusterDomainName(), err)
+	}
+	return &api.SdkClusterDomainActivateResponse{}, nil
+}
+
+func (m *ClusterDomainsServer) Deactivate(
+	ctx context.Context,
+	req *api.SdkClusterDomainDeactivateRequest,
+) (*api.SdkClusterDomainDeactivateResponse, error) {
+	if m.cluster() == nil {
+		return nil, status.Error(codes.Unavailable, "Resource has not been initialized")
+	}
+
+	if !api.IsAdminByContext(ctx) {
+		return nil, status.Errorf(codes.PermissionDenied, "Activating cluster domain %v not permitted", req.GetClusterDomainName())
```

Comment:
may want to add to the error that `only the storage administrator is authorized`